go/libraries/doltcore/sqle/commit_diff_table.go

```diff
@@ -0,0 +1,397 @@
+
+	if err != nil {
+		return nil, err
+	}
+
+	fromTable, _, err := fromRoot.GetTable(ctx, dt.name)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return NewSliceOfPartitionsItr([]sql.Partition{diffPartition{
+		to:       toTable,
+		from:     fromTable,
+		toName:   toName,
+		fromName: fromName,
+		toDate:   toDate,
+		fromDate: fromDate,
+	}}), nil
+}
+
+func (dt *CommitDiffTable) rootValForFilter(ctx *sql.Context, eqFilter *expression.Equals) (*doltdb.RootValue, string, *types.Timestamp, error) {
+	var hashStr string
+	var ok bool
+	if _, isGetField := eqFilter.BinaryExpression.Left.(*expression.GetField); !isGetField {
+		val, err := eqFilter.BinaryExpression.Left.Eval(ctx, nil)
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		if hashStr, ok = val.(string); !ok {
+			return nil, "", nil, fmt.Errorf("received '%v' when expecting commit hash string", val)
+		}
+	} else if _, isGetField = eqFilter.BinaryExpression.Right.(*expression.GetField); !isGetField {
+		val, err := eqFilter.BinaryExpression.Right.Eval(ctx, nil)
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		if hashStr, ok = val.(string); !ok {
+			return nil, "", nil, fmt.Errorf("received '%v' when expecting commit hash string", val)
+		}
+	}
+
+	var root *doltdb.RootValue
+	var commitTime *types.Timestamp
+	if strings.ToLower(hashStr) == "working" {
+		root = dt.workingRoot
+	} else {
+		cs, err := doltdb.NewCommitSpec(hashStr)
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		cm, err := dt.ddb.Resolve(ctx, cs, nil)
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		root, err = cm.GetRootValue()
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		meta, err := cm.GetCommitMeta()
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		t := meta.Time()
+		commitTime = (*types.Timestamp)(&t)
+	}
+
+	return root, hashStr, commitTime, nil
+}
+
+// HandledFilters returns the list of filters that will be handled by the table itself
+func (dt *CommitDiffTable) HandledFilters(filters []sql.Expression) []sql.Expression {
+	var remaining []sql.Expression
+	for _, filter := range filters {
+		isCommitFilter := false
+
+		if eqFilter, isEquality := filter.(*expression.Equals); isEquality {
+			sql.Inspect(filter, func(e sql.Expression) (cont bool) {
+				if e == nil {
+					return true
+				}
+
+				switch val := e.(type) {
+				case *expression.GetField:
+					switch strings.ToLower(val.Name()) {
+					case toCommit:
+						if dt.toCommitFilter != nil {
+							dt.requiredFilterErr = ErrExactlyOneToCommit
+						}
+
+						isCommitFilter = true
+						dt.toCommitFilter = eqFilter
+					case fromCommit:
+						if dt.fromCommitFilter != nil {
+							dt.requiredFilterErr = ErrExactlyOneFromCommit
+						}
+
+						isCommitFilter = true
+						dt.fromCommitFilter = eqFilter
+					}
+				}
+
+				return true
+			})
+		}
+
+		if !isCommitFilter {
+			remaining = append(remaining, filter)
+		}
+	}
+
+	return remaining
```

Comment:
Isn't this the opposite logic you want? I thought this interface returned the set of expressions handled. Am I way off base?

Reply:
updated tests and fixed